### PR TITLE
allow adding tags as a custom subject format for emails

### PR DESCRIPTION
**Pull request by majakomel**
Meta: https://meta.discourse.org/t/allow-adding-tags-as-a-custom-subject-format-for-emails/84424

### CODE REVIEW

File: app/mailers/user_notifications.rb [hunk 1]
```diff
@@ -433,6 +443,14 @@ def send_notification_email(opts)
       show_category_in_subject = nil
     end
 
+    # tag names
+    tags = Topic.find_by(id: post.topic_id)&.tags
+    if opts[:show_tags_in_subject] && post.topic_id && tags
```
Comment (ZogStriP) [inline app/mailers/user_notifications.rb]: I would first check tags are allowed in subject and only then do the SQL query.

Comment (ZogStriP) [reply app/mailers/user_notifications.rb]: Also, there should probably be a limit of the number of tags retrieved.

Comment (majakomel) [reply app/mailers/user_notifications.rb]: What would be the best limit? 3 tags? Should there be any specific order as well then?

Comment (ZogStriP) [reply app/mailers/user_notifications.rb]: 3 seems fine. I would keep the same order as in the post.

**Comment by SamSaffron** [comment]
This looks good to me, should we be adding spaces between tags or having them this close together? 

Note we need to wait 2 weeks to merge this in cause we are stabalizing the 2.0 release at the moment.

**Comment by SamSaffron** [comment]
cool, merging now

**Pull request merged**